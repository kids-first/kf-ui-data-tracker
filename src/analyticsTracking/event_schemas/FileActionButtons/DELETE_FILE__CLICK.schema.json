{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/FileActionButtons/DELETE_FILE__CLICK.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "DELETE_FILE__CLICK",
  "type": "object",
  "description": "click event for 'Delete' button inside of FileActionButtons 'Are you sure?' toolip",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../BUTTON/button_defintions.json#/link"},
    {"$ref": "../TOOLTIP/tooltip_defintions.json#/tooltip_content"},
    {
      "properties": {
        "file": {
          "type": "object",
          "description": "study document to be deleted",
          "properties": {
            "studyId": {
              "type": "string",
              "format": "(SD)[aZ09]{8}",
              "description": "kids first study identifier"
            },
            "kfId": {
              "type": "string",
              "format": "(SF_)[aZ09]{8}",
              "description": "'SF_' prefixed kids first file id for the study of which the file is in"
            }
          }
        },
        "is_admin": {
          "type": "boolean",
          "description": "whether or not the current use is in the ADMIN Auth0 group"
        }
      }
    }
  ]
}
