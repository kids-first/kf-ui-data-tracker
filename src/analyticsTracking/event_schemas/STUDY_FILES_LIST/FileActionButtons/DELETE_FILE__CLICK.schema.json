{
  "$id": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/STUDY_FILES_LIST/FileActionButtons/DELETE_FILE__CLICK.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "DELETE_FILE__CLICK",
  "type": "object",
  "description": "click event for 'Delete' button inside of FileActionButtons 'Are you sure?' toolip",
  "properties": {
    "path": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/common_defs.schema.json#/definitions/path"
    },
    "route": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/common_defs.schema.json#/definitions/route"
    },
    "view": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/common_defs.schema.json#/definitions/view"
    },
    "study": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/common_defs.schema.json#/definitions/study"
    },
    "scope": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/common_defs.schema.json#/definitions/scope"
    },
    "link": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/BUTTON/BUTTON_EVENT.schema.json#/definitions/link"
    },
    "button_text": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/BUTTON/BUTTON_EVENT.schema.json#/definitions/button_text"
    },
    "button_type": {
      "$ref": "http://kf-ui-data-tracker.kidsfirstdrc.org/src/analyticsTracking/event_schemas/BUTTON/BUTTON_EVENT.schema.json#/definitions/button_type"
    },
    "file": {
      "type": "object",
      "description": "study document to be deleted",
      "properties": {
        "studyId": {
          "type": "string",
          "pattern": "SD_[A-Z0-9]{8}",
          "description": "kids first study identifier"
        },
        "kfId": {
          "type": "string",
          "pattern": "SF_[A-Z0-9]{8}",
          "description": "'SF_' prefixed kids first file id for the study of which the file is in"
        }
      }
    },
    "is_admin": {
      "type": "boolean",
      "description": "whether or not the current use is in the ADMIN Auth0 group"
    }
  },
  "required": ["path", "route", "view", "file", "is_admin"],
  "additionalProperties": false
}
