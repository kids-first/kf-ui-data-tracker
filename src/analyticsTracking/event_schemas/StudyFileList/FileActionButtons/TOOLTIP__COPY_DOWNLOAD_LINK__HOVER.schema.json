{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/TOOLTIP/TOOLTIP__COPY_DOWNLOAD_LINK__HOVER.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "TOOLTIP__COPY_DOWNLOAD_LINK__HOVER",
  "type": "object",
  "description": "",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../TOOLTIP/tooltip_defintions.json#/tooltip_name"},
    {"$ref": "../TOOLTIP/tooltip_defintions.json#/tooltip_content"},
    {
      "properties": {
        "file": {
          "type": "object",
          "description": "",
          "properties": {
            "studyId": {
              "type": "string",
              "format": "(SD)[aZ09]{8}",
              "description": "kids first study identifier"
            },
            "kfId": {
              "type": "string",
              "format": "(SF_)[aZ09]{8}",
              "description": "'SF_' prefixed kids first file id for the study of which the file is in"
            }
          }
        },
        "is_admin": {
          "type": "boolean",
          "description": "whether or not the current use is in the ADMIN Auth0 group"
        },
        "link": {
          "type": "null",
          "description": "download url of most recent file version in the format of `download/study/<kf_study_id>/file/<kf_file_id>/version/<kf_study_file_version_id>`"
        }
      }
    }
  ]
}
