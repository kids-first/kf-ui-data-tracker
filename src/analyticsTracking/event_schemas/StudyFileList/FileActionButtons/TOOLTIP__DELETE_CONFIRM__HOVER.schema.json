{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/FileActionButtons/TOOLTIP__DELETE_CONFIRM__HOVER.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "TOOLTIP__DELETE_CONFIRM__HOVER",
  "type": "object",
  "description": "mouse over event for FileActionButtons delete icon and tooltip",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../BUTTON/button_defintions.json#/link"},
    {"$ref": "../TOOLTIP/tooltip_defintions.json#/tooltip_name"},
    {"$ref": "../TOOLTIP/tooltip_defintions.json#/tooltip_content"},
    {
      "properties": {
        "file": {
          "type": "object",
          "description": "",
          "properties": {
            "studyId": {
              "type": "string",
              "format": "(SD)[aZ09]{8}",
              "description": "kids first study identifier"
            },
            "kfId": {
              "type": "string",
              "format": "(SF_)[aZ09]{8}",
              "description": "'SF_' prefixed kids first file id for the study of which the file is in"
            }
          }
        },
        "is_admin": {
          "type": "boolean",
          "description": "whether or not the current use is in the ADMIN Auth0 group"
        }
      }
    }
  ]
}
