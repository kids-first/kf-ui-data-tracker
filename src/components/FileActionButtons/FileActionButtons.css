.FileActions {
  @inherit: .w-1\/2, .block, .text-center, .bg-white, .py-8;
  justify-self: end;
}

.FileActions-Button {
  @inherit: .w-auto, .mx-auto, .py-4, .block, .text-center;
}

.FileActions-Button:focus {
  opacity: 1;
}
