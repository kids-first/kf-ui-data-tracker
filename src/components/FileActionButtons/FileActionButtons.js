import React from 'react';
import PropTypes from 'prop-types';
import {Button, Divider, Icon, Popup} from 'semantic-ui-react';
import {downloadFile} from '../../common/fileUtils';
/**
 * Simple Icon Buttons to download and delete a file passed in as the node prop
 */
const FileActionButtons = ({
  node,
  loading,
  error,
  studyId,
  downloadFileMutation,
  deleteFile,
  vertical = true,
  fluid = false,
}) => {
  return (
    <Button.Group vertical={vertical} fluid={fluid} size="mini">
      <Button
        basic
        icon="download"
        onClick={e => {
          e.stopPropagation();
          downloadFile(studyId, node.kfId, null, downloadFileMutation);
        }}
      />
      <Popup
        trigger={
          <Button
            basic
            data-testid="delete-button"
            onClick={e => e.stopPropagation()}
            icon={<Icon name="trash alternate" />}
          />
        }
        header="Are you sure?"
        content={
          <>
            This file and all of its versions and history will be deleted
            <Divider />
            <Button
              data-testid="delete-confirm"
              negative
              fluid
              icon={<Icon name="trash alternate" />}
              content="Delete"
              onClick={e => {
                e.stopPropagation();
                deleteFile({variables: {kfId: node.kfId}});
              }}
            />
          </>
        }
        on="click"
        position="top right"
      />
    </Button.Group>
  );
};

FileActionButtons.propTypes = {
  /** study file object */
  node: PropTypes.object.isRequired,
  /** loading state */
  loading: PropTypes.bool,
  /** graphQL errors */
  error: PropTypes.object,
  /** Kids First unique study identifier (SD_XXXXXXXX) */
  studyId: PropTypes.string.isRequired,
  /** Action to delete a file */
  deleteFile: PropTypes.func,
  /** Action to download a file */
  downloadFile: PropTypes.func,
  /** Whether the buttons should be grouped vertically or not */
  vertical: PropTypes.bool,
  /** Whether the buttons should fill their container or not */
  fluid: PropTypes.bool,
};

export default FileActionButtons;
