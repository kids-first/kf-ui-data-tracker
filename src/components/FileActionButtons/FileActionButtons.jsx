import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {KF_STUDY_API} from '../../common/globals';
import {Icon} from 'kf-uikit';

/**
 * Simple Icon Buttons to download and delete a file passed in as the node prop
 */
const FileActionButtons = ({
  node,
  loading,
  error,
  studyId,
  downloadFile,
  deleteFile,
  className,
  buttonClass,
}) => {
  const FileActionsClass = classes(
    'FileActions',
    {'cursor-not-allowed': loading || error},
    className,
  );
  const FileActionButton = classes('FileActions-Button', buttonClass);

  return (
    <div className={FileActionsClass}>
      <button
        className={FileActionButton}
        onClick={e => {
          downloadFile({
            variables: {studyId, fileId: node.kfId},
          }).then(resp => {
            const url = `${KF_STUDY_API}${resp.data.signedUrl.url}`;
            window.location.href = url;
          });
        }}
      >
        <Icon color="black" width={16} height={16} kind="download" />
      </button>
      <button
        className={FileActionButton}
        onClick={e => deleteFile({variables: {kfId: node.kfId}})}
      >
        <Icon width={16} height={16} kind="delete" />
      </button>
    </div>
  );
};

FileActionButtons.propTypes = {
  /** study file object */
  node: PropTypes.object.isRequired,
  /** loading state */
  loading: PropTypes.bool,
  /** graphQL errors */
  error: PropTypes.object,
  /** Kids First unique study identifier (SD_XXXXXXXX) */
  studyId: PropTypes.string.isRequired,
  /** Action to delete a file */
  deleteFile: PropTypes.func,
  /** Action to download a file */
  downloadFile: PropTypes.func,
  /** additional classes to add to the container element */
  className: PropTypes.string,
  /** additional classes to add to the buttons */
  buttonClass: PropTypes.string,
};

export default FileActionButtons;
