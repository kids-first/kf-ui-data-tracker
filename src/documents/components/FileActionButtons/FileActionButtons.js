import React from 'react';
import PropTypes from 'prop-types';
import {Amplitude} from '@amplitude/react-amplitude';
import {Button, Divider, Icon, Popup, Responsive} from 'semantic-ui-react';
import {downloadFile, fileSortedVersions} from '../../utilities';
import CopyButton from '../../../components/CopyButton/CopyButton';
/**
 * Simple Icon Buttons to download and delete a file passed in as the node prop
 */
const FileActionButtons = ({
  node,
  loading,
  error,
  studyId,
  downloadFileMutation,
  deleteFile,
  vertical = true,
  fluid = false,
  hideCopy = false,
  version = null,
  updateReview,
  reviewNode,
  updateError,
}) => {
  return (
    <Button.Group fluid={fluid} size="small">
      <Amplitude
        eventProperties={inheritedProps => ({
          ...inheritedProps,
          scope: inheritedProps.scope
            ? [...inheritedProps.scope, 'button', 'preview button']
            : ['button', 'preview button'],
        })}
      >
        {({logEvent}) => (
          <Popup
            inverted
            position="top left"
            content={version ? 'Preview version' : 'Preview latest version'}
            trigger={
              <Button
                basic
                compact
                icon="eye"
                onClick={e => {
                  logEvent('click');
                  e.stopPropagation();
                  e.preventDefault();
                  window.open(
                    `/study/${studyId}/documents/${node.kfId}/versions/${
                      version
                        ? version.kfId
                        : fileSortedVersions(node)[0].node.kfId
                    }`,
                  );
                }}
              />
            }
          />
        )}
      </Amplitude>
      <Amplitude
        eventProperties={inheritedProps => ({
          ...inheritedProps,
          scope: inheritedProps.scope
            ? [...inheritedProps.scope, 'button', 'download button']
            : ['button', 'download button'],
        })}
      >
        {({logEvent}) => (
          <Popup
            inverted
            position="top left"
            icon="download"
            content={version ? 'Download version' : 'Download latest version'}
            trigger={
              <Button
                as="a"
                href={
                  node.downloadUrl +
                  `/version/${
                    version
                      ? version.kfId
                      : fileSortedVersions(node)[0].node.kfId
                  }`
                }
                basic
                compact
                icon="download"
                data-testid="download-file"
                onClick={e => {
                  logEvent('click');
                  e.stopPropagation();
                  e.preventDefault();
                  downloadFile(
                    studyId,
                    node.kfId,
                    version && version.kfId,
                    downloadFileMutation,
                  );
                }}
              />
            }
          />
        )}
      </Amplitude>
      {!hideCopy && (
        <Amplitude
          eventProperties={inheritedProps => ({
            ...inheritedProps,
            scope: inheritedProps.scope
              ? [...inheritedProps.scope, 'button', 'copy ID button']
              : ['button', 'copy ID button'],
          })}
        >
          {({logEvent}) => (
            <CopyButton
              data-testid="copy-file-id"
              textToCopy={
                node.downloadUrl +
                `/version/${
                  version ? version.kfId : fileSortedVersions(node)[0].node.kfId
                }`
              }
              basic
              compact
              position="top left"
              tooltip="Copy download link"
              onClick={e => {
                logEvent('click');
                e.stopPropagation();
              }}
            />
          )}
        </Amplitude>
      )}
      {deleteFile && (
        <Amplitude
          eventProperties={inheritedProps => ({
            ...inheritedProps,
            scope: inheritedProps.scope
              ? [...inheritedProps.scope, 'button', 'delete button']
              : ['button', 'delete button'],
          })}
        >
          {({logEvent}) => (
            <Popup
              trigger={
                <Responsive
                  as={Button}
                  minWidth={Responsive.onlyTablet.minWidth}
                  basic
                  compact
                  data-testid="delete-button"
                  onClick={e => e.stopPropagation()}
                  icon={<Icon name="trash alternate" />}
                />
              }
              header="Are you sure?"
              content={
                <>
                  This file and all of its versions and history will be deleted
                  <Divider />
                  <Button
                    data-testid="delete-confirm"
                    negative
                    fluid
                    icon={<Icon name="trash alternate" />}
                    content="Delete"
                    onClick={e => {
                      logEvent('click');
                      e.stopPropagation();
                      deleteFile({variables: {kfId: node.kfId}});
                      window.location.reload(false);
                    }}
                  />
                </>
              }
              on="click"
              position="top right"
            />
          )}
        </Amplitude>
      )}
      {version &&
        updateReview &&
        reviewNode &&
        reviewNode.state !== 'completed' &&
        reviewNode.state !== 'closed' && (
          <Amplitude
            eventProperties={inheritedProps => ({
              ...inheritedProps,
              scope: inheritedProps.scope
                ? [...inheritedProps.scope, 'button', 'remove version button']
                : ['button', 'remove version button'],
            })}
          >
            {({logEvent}) => (
              <Popup
                trigger={
                  <Responsive
                    as={Button}
                    minWidth={Responsive.onlyTablet.minWidth}
                    basic
                    compact
                    onClick={e => e.stopPropagation()}
                    icon={<Icon name="minus square" color="red" />}
                  />
                }
                header="Are you sure?"
                content={
                  <>
                    This version will be removed from current data review
                    <Divider />
                    {updateError ? (
                      <span className="text-red">{updateError.message}</span>
                    ) : (
                      <Button
                        negative
                        fluid
                        icon={<Icon name="minus square" />}
                        content="Remove"
                        onClick={e => {
                          logEvent('click');
                          e.stopPropagation();
                          const versionsList = reviewNode.versions.edges.map(
                            ({node}) => node.id,
                          );
                          updateReview({
                            variables: {
                              id: reviewNode.id,
                              input: {
                                name: reviewNode.name,
                                description: reviewNode.descrition,
                                versions: versionsList.filter(
                                  id => id !== version.id,
                                ),
                              },
                            },
                          }).catch(err => console.log(err));
                        }}
                      />
                    )}
                  </>
                }
                on="click"
                position="top right"
              />
            )}
          </Amplitude>
        )}
    </Button.Group>
  );
};

FileActionButtons.propTypes = {
  /** study file object */
  node: PropTypes.object.isRequired,
  /** loading state */
  loading: PropTypes.bool,
  /** graphQL errors */
  error: PropTypes.object,
  /** Kids First unique study identifier (SD_XXXXXXXX) */
  studyId: PropTypes.string.isRequired,
  /** Action to delete a file */
  deleteFile: PropTypes.func,
  /** Action to download a file */
  downloadFileMutation: PropTypes.func,
  /** Whether the buttons should be grouped vertically or not */
  vertical: PropTypes.bool,
  /** Whether the buttons should fill their container or not */
  fluid: PropTypes.bool,
};

export default FileActionButtons;
