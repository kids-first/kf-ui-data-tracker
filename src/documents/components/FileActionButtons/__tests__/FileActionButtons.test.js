import React from 'react';
import wait from 'waait';
import {render, cleanup, fireEvent, act} from 'react-testing-library';
import {MockedProvider} from 'react-apollo/test-utils';
import {mocks} from '../../../../../__mocks__/kf-api-study-creator/mocks';
import studyByKfId from '../../../../../__mocks__/kf-api-study-creator/responses/studyByKfId';
import FileActionButtons from '../FileActionButtons';

afterEach(cleanup);
it('File Action Buttons renders correctly -- default look', async () => {
  const studyId = studyByKfId.data.studyByKfId.kfId;
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MockedProvider mocks={mocks}>
      <FileActionButtons
        node={file}
        studyId={studyId}
        downloadFileMutation={jest.fn().mockResolvedValue({
          data: {
            signedUrl: {
              url:
                'http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM16',
            },
          },
        })}
        deleteFile={jest.fn()}
      />
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  // Click on the download file button
  act(() => {
    fireEvent.click(tree.getAllByTestId('download-file')[0]);
  });
  await wait();

  expect(tree.container).toMatchSnapshot();

  // Click on the delete file button
  act(() => {
    fireEvent.click(tree.getAllByTestId('delete-button')[0]);
  });
  await wait();

  expect(tree.container).toMatchSnapshot();
});

it('File Action Buttons renders correctly -- vertical look', () => {
  const studyId = studyByKfId.data.studyByKfId.kfId;
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MockedProvider mocks={mocks}>
      <FileActionButtons
        node={file}
        studyId={studyId}
        downloadFileMutation={jest.fn()}
        deleteFile={jest.fn()}
        vertical
      />
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('File Action Buttons renders correctly -- fluid look', () => {
  const studyId = studyByKfId.data.studyByKfId.kfId;
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <MockedProvider mocks={mocks}>
      <FileActionButtons
        node={file}
        studyId={studyId}
        downloadFileMutation={jest.fn()}
        deleteFile={jest.fn()}
        fluid
      />
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();
});
