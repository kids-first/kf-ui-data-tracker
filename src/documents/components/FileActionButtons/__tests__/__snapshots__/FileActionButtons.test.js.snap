// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Action Buttons renders correctly -- default look 1`] = `
<div>
  <div
    class="ui small buttons"
  >
    <button
      class="ui basic compact icon button"
      data-testid="copy-file-id"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
    </button>
    <a
      class="ui basic compact icon button"
      data-testid="download-file"
      href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
      role="button"
    >
      <i
        aria-hidden="true"
        class="download icon"
      />
    </a>
    <button
      class="ui basic compact icon button"
      data-testid="delete-button"
    >
      <i
        aria-hidden="true"
        class="trash alternate icon"
      />
    </button>
  </div>
</div>
`;

exports[`File Action Buttons renders correctly -- default look 2`] = `
<div>
  <div
    class="ui small buttons"
  >
    <button
      class="ui basic compact icon button"
      data-testid="copy-file-id"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
    </button>
    <a
      class="ui basic compact icon button"
      data-testid="download-file"
      href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
      role="button"
    >
      <i
        aria-hidden="true"
        class="download icon"
      />
    </a>
    <button
      class="ui basic compact icon button"
      data-testid="delete-button"
    >
      <i
        aria-hidden="true"
        class="trash alternate icon"
      />
    </button>
  </div>
</div>
`;

exports[`File Action Buttons renders correctly -- default look 3`] = `
<div>
  <div
    class="ui small buttons"
  >
    <button
      class="ui basic compact icon button"
      data-testid="copy-file-id"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
    </button>
    <a
      class="ui basic compact icon button"
      data-testid="download-file"
      href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
      role="button"
    >
      <i
        aria-hidden="true"
        class="download icon"
      />
    </a>
    <button
      class="ui basic compact icon button"
      data-testid="delete-button"
    >
      <i
        aria-hidden="true"
        class="trash alternate icon"
      />
    </button>
  </div>
</div>
`;

exports[`File Action Buttons renders correctly -- fluid look 1`] = `
<div>
  <div
    class="ui small fluid buttons"
  >
    <button
      class="ui basic compact icon button"
      data-testid="copy-file-id"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
    </button>
    <a
      class="ui basic compact icon button"
      data-testid="download-file"
      href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
      role="button"
    >
      <i
        aria-hidden="true"
        class="download icon"
      />
    </a>
    <button
      class="ui basic compact icon button"
      data-testid="delete-button"
    >
      <i
        aria-hidden="true"
        class="trash alternate icon"
      />
    </button>
  </div>
</div>
`;

exports[`File Action Buttons renders correctly -- vertical look 1`] = `
<div>
  <div
    class="ui small buttons"
  >
    <button
      class="ui basic compact icon button"
      data-testid="copy-file-id"
    >
      <i
        aria-hidden="true"
        class="copy icon"
      />
    </button>
    <a
      class="ui basic compact icon button"
      data-testid="download-file"
      href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
      role="button"
    >
      <i
        aria-hidden="true"
        class="download icon"
      />
    </a>
    <button
      class="ui basic compact icon button"
      data-testid="delete-button"
    >
      <i
        aria-hidden="true"
        class="trash alternate icon"
      />
    </button>
  </div>
</div>
`;
